allennlp/models/simple_tagger.py

```diff
@@ -141,6 +135,16 @@ def tag(self, text_field: TextField) -> Dict[str, Any]:
         predictions = output_dict["class_probabilities"].data.squeeze(0)
         _, argmax = predictions.max(-1)
         indices = argmax.squeeze(1).numpy()
-        tags = [self.vocabulary.get_token_from_index(x, namespace="tags") for x in indices]
+        tags = [self.vocab.get_token_from_index(x, namespace="tags") for x in indices]
 
         return {"tags": tags, "class_probabilities": predictions.numpy()}
+
+    @classmethod
+    def from_params(cls, vocab: Vocabulary, params: Params) -> 'SimpleTagger':
+        hidden_size = params.pop("hidden_size", 200)
+        num_layers = params.pop("num_layers", 2)
+        token_embedder = TokenEmbedder.from_params(vocab, params.pop("token_embedder"))
+        return cls(vocab=vocab,
```

Comment:
I believe if you do

```
class ComplicatedTagger(SimpleTagger):
  ...
```

then `ComplicatedTagger.from_params` returns an instance of `ComplicatedTagger` (assuming `ComplicatedTagger` has the same constructor signature, otherwise it might crash). I'm not sure sure how `mypy` would feel about that, you might have to use a `cast` afterward.